core-rust/state-manager/src/state_manager.rs

```diff
@@ -127,31 +129,42 @@ impl StateManagerConfig {
             state_hash_tree_gc_config: StateHashTreeGcConfig::default(),
             ledger_proofs_gc_config: LedgerProofsGcConfig::default(),
             ledger_sync_limits_config: LedgerSyncLimitsConfig::default(),
+            protocol_config: ProtocolConfig::no_updates("testing-genesis".to_string()),
             no_fees: false,
         }
     }
 }
 
 #[derive(Clone)]
 pub struct StateManager {
+    config: StateManagerConfig,
     pub state_computer: Arc<StateComputer<StateManagerDatabase>>,
     pub database: Arc<StateLock<StateManagerDatabase>>,
     pub pending_transaction_result_cache: Arc<RwLock<PendingTransactionResultCache>>,
     pub mempool: Arc<RwLock<PriorityMempool>>,
     pub mempool_manager: Arc<MempoolManager>,
-    pub committability_validator: Arc<CommittabilityValidator<StateManagerDatabase>>,
-    pub transaction_previewer: Arc<TransactionPreviewer<StateManagerDatabase>>,
+    pub execution_configurator: Arc<RwLock<ExecutionConfigurator>>,
+    pub committability_validator: Arc<RwLock<CommittabilityValidator<StateManagerDatabase>>>,
+    pub transaction_previewer: Arc<RwLock<TransactionPreviewer<StateManagerDatabase>>>,
+    pub protocol_updater_factory: Arc<dyn ProtocolUpdaterFactory + Sync + Send>,
```

Comment:
It doesn't seem like this needs to be `dyn`, is it possible to just use a generic?